src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -144,11 +132,11 @@ protected void move(final boolean nonCombat, final IMoveDelegate moveDel, final
     initializeData();
     ProBattleUtils.setData(data);
     if (nonCombat) {
-      nonCombatMoveAI.doNonCombatMove(storedFactoryMoveMap, storedPurchaseTerritories, moveDel, data, player, false);
+      nonCombatMoveAI.doNonCombatMove(storedFactoryMoveMap, storedPurchaseTerritories, moveDel, false);
```

Comment:
Down to 4 args now from 6, that is huge. 4 is still a lot, considering the number of permutations to test that, clearly 4 arg methods are to be avoided. Would be nice to see the boolean parameter dropped in one manner or another as well, but this is good progress otherwise.